src/os/bluestore/BlueFS.h

```diff
@@ -459,6 +459,11 @@ class BlueFS {
     _close_writer(h);
   }
 
+  void close_reader(FileReader *h) {
+    std::lock_guard l(lock);
```

Comment:
why do we need to hold a lock for deleting `FileReader`?

Reply:
when we called the function such as "BlueFS::open_for_read", we need to hold a lock, so need to hold a lock for deleting FileReader.

Reply:
i am not convinced.